components/sidebar/sidebar_channel/sidebar_channel_menu/sidebar_channel_menu.tsx

```diff
@@ -13,292 +22,241 @@ import ChannelInviteModal from 'components/channel_invite_modal';
-                    text={intl.formatMessage({id: 'sidebar_left.sidebar_channel_menu.markAsUnread', defaultMessage: 'Mark as Unread'})}
-                />
-            );
-        }
+    function onToggleMenu(open: boolean) {
+        props.onToggleMenu(open);
 
-        let favorite;
-        if (isFavorite) {
-            favorite = (
-                <Menu.ItemAction
-                    id={`unfavorite-${channel.id}`}
-                    onClick={this.unfavoriteChannel}
-                    icon={<i className='icon-star'/>}
-                    text={intl.formatMessage({id: 'sidebar_left.sidebar_channel_menu.unfavoriteChannel', defaultMessage: 'Unfavorite'})}
-                />
-            );
-        } else {
-            favorite = (
-                <Menu.ItemAction
-                    id={`favorite-${channel.id}`}
-                    onClick={this.favoriteChannel}
-                    icon={<i className='icon-star-outline'/>}
-                    text={intl.formatMessage({id: 'sidebar_left.sidebar_channel_menu.favoriteChannel', defaultMessage: 'Favorite'})}
-                />
-            );
+        if (open) {
+            trackEvent('ui', 'ui_sidebar_channel_menu_opened');
         }
+    }
 
-        let muteChannel;
-        if (isMuted) {
-            let muteChannelText = intl.formatMessage({id: 'sidebar_left.sidebar_channel_menu.unmuteChannel', defaultMessage: 'Unmute Channel'});
-            if (channel.type === Constants.DM_CHANNEL || channel.type === Constants.GM_CHANNEL) {
-                muteChannelText = intl.formatMessage({id: 'sidebar_left.sidebar_channel_menu.unmuteConversation', defaultMessage: 'Unmute Conversation'});
-            }
-            muteChannel = (
-                <Menu.ItemAction
-                    id={`unmute-${channel.id}`}
-                    onClick={this.unmuteChannel}
-                    icon={<i className='icon-bell-off-outline'/>}
-                    text={muteChannelText}
-                />
-            );
-        } else {
-            let muteChannelText = intl.formatMessage({id: 'sidebar_left.sidebar_channel_menu.muteChannel', defaultMessage: 'Mute Channel'});
-            if (channel.type === Constants.DM_CHANNEL || channel.type === Constants.GM_CHANNEL) {
-                muteChannelText = intl.formatMessage({id: 'sidebar_left.sidebar_channel_menu.muteConversation', defaultMessage: 'Mute Conversation'});
-            }
-            muteChannel = (
-                <Menu.ItemAction
-                    id={`mute-${channel.id}`}
-                    onClick={this.muteChannel}
-                    icon={<i className='icon-bell-outline'/>}
-                    text={muteChannelText}
-                />
-            );
-        }
+    let markAsReadMenuItem: JSX.Element | null = null;
+    if (props.isUnread) {
+        markAsReadMenuItem = (
+            <Menu.ItemAction
+                id={`markAsRead-${props.channel.id}`}
+                onClick={handleMarkAsRead}
```

Comment:
If you go in the direction of re-creating the function on every render, you could add it as an anonymous function here, so the function only gets created in memory when it is actually needed. (And similar with all the similar elements)

Reply:
changed